sycl/doc/extensions/ExtendedAtomics/SYCL_INTEL_extended_atomics.asciidoc

```diff
@@ -113,389 +115,399 @@ In SYCL 1.2.1, support for floating-point types is limited to the +load+, +store
+class atomic_ref {
  public:
 
-  atomic(multi_ptr<T, addressSpace> ptr);
-  atomic(const atomic&);
-  atomic& operator=(const atomic&) = delete;
+  using value_type = T;
+  static constexpr size_t required_alignment = /* implementation-defined */;
+  static constexpr bool is_always_lock_free = /* implementation-defined */;
+  static constexpr memory_order default_read_order = memory_order_traits<DefaultOrder>::read_order;
+  static constexpr memory_order default_write_order = memory_order_traits<DefaultOrder>::write_order;
+  static constexpr memory_order default_read_modify_write_order = DefaultOrder;
 
   bool is_lock_free() const;
 
-  void store(T operand, memory_order order =
-    memory_order::relaxed);
+  explicit atomic_ref(T&);
+  atomic_ref(const atomic_ref&);
+  atomic_ref& operator=(const atomic_ref&) = delete;
 
-  T operator=(T desired);
+  void store(T operand,
+    memory_order order = default_write_order) const;
 
-  T load(memory_order order = memory_order::relaxed) const;
+  T operator=(T desired) const;
+
+  T load(memory_order order = default_read_order) const;
 
   operator T() const;
 
-  T exchange(T operand, memory_order order =
-    memory_order::relaxed);
+  T exchange(T operand,
+    memory_order order = default_read_modify_write_order) const;
 
   bool compare_exchange_weak(T &expected, T desired,
-    memory_order success = memory_order::relaxed,
-    memory_order failure = memory_order::relaxed);
+    memory_order success = default_read_modify_write_order,
+    memory_order failure = default_read_order) const;
 
   bool compare_exchange_weak(T &expected, T desired,
-    memory_order order = memory_order::relaxed);
+    memory_order order = default_read_modify_write_order) const;
 
   bool compare_exchange_strong(T &expected, T desired,
-    memory_order success = memory_order::relaxed,
-    memory_order failure = memory_order::relaxed);
+    memory_order success = default_read_modify_write_order,
+    memory_order failure = default_read_order) const;
 
   bool compare_exchange_strong(T &expected, T desired,
-    memory_order order = memory_order::relaxed);
-
-  T fetch_add(T operand, memory_order order =
-    memory_order::relaxed);
-
-  T fetch_sub(T operand, memory_order order =
-    memory_order::relaxed);
+    memory_order order = default_read_modify_write_order) const;
+};
 
-  T fetch_min(T operand, memory_order order =
-    memory_order::relaxed);
+// Partial specialization for integral types
+template <memory_order DefaultOrder, access::address_space Space>
+class atomic_ref<Integral, DefaultOrder, Space> {
 
-  T fetch_max(T operand, memory_order order =
-    memory_order::relaxed);
+  /* All other members from atomic_ref<T> are available */
 
-  T operator+=(T operand);
-  T operator-=(T operand);
+  using difference_type = value_type;
 
-  /* Available only when T is Integral */
-  T fetch_and(T operand, memory_order order =
-    memory_order::relaxed);
+  Integral fetch_add(Integral operand,
```

Comment:
Why `operator+=` is marked as `noexcept`, while `fetch_add` isn't?

Reply:
They should all be `const noexcept`, and I missed some by accident.  Fixed.